src/markerclusterer.ts

```diff
@@ -178,8 +178,22 @@ export class MarkerClusterer extends OverlayViewSafe {
 
       // allow algorithms to return flag on whether the clusters/markers have changed
       if (changed || changed == undefined) {
+        clusters.forEach((cluster, index) => {
+          // if there's clusters that has been "processed" (Cluster.marker is set), then reuse it instead of reprocessing it
+          const duplicate = this.clusters.findIndex((old) => old.marker && old.marker.getPosition().equals(cluster.position));
```

Comment:
It looks like the only change to support advanced markers would be to use `MarkerUtils.getPosition()` here

https://github.com/googlemaps/js-markerclusterer/blob/9476b3a62399fd124c92ed74faae6943cf573ea4/src/marker-utils.ts#L41-L60